test/Abp.ZeroCore.Tests/Zero/EntityHistory/SimpleEntityHistory_Test.cs

```diff
@@ -274,38 +273,97 @@ public void Should_Write_History_For_Audited_Entities_Update()
         }
 
         [Fact]
-        public void Should_Write_History_For_Audited_Entities_Update_Owned()
+        public void Should_Write_History_For_Owned_Entity_Of_Audited_Entities_Create()
         {
-            /* Blog has Audited attribute. */
+            // Blog is the owner of BlogEx and has Audited attribute.
+            // Therefore, BlogEx should follow Blog and have entity history.
 
-            int blog1Id;
-            var newValue = "blogger-2";
-            string originalValue;
+            WithUnitOfWork(() =>
+            {
+                // since owned entities is not available via DbContext -> DbSet,
+                // we have to load it via it's owner entity
+                var blog1 = _blogRepository.Single(b => b.Name == "test-blog-2");
+                blog1.More.ShouldBeNull();
+            });
 
-            using (var uow = Resolve<IUnitOfWorkManager>().Begin())
+            WithUnitOfWork(() =>
             {
+                var blog1 = _blogRepository.Single(b => b.Name == "test-blog-2");
+                blog1.More = new BlogEx() { BloggerName = "blogger-2" };
+            });
+
+            WithUnitOfWork(() =>
+            {
+                // since owned entities is not available via DbContext -> DbSet,
+                // we have to load it via it's owner entity
+                var blog1 = _blogRepository.Single(b => b.Name == "test-blog-2");
+                blog1.More.ShouldNotBeNull();
+            });
+
+            Predicate<EntityChangeSet> predicate = s =>
+            {
+                s.EntityChanges.Count.ShouldBe(1);
+
+                var entityChange = s.EntityChanges.Single(ec => ec.EntityTypeFullName == typeof(BlogEx).FullName);
+                entityChange.ChangeType.ShouldBe(EntityChangeType.Created);
+                // The primary key of BlogEx is a shadow property,
+                // and we assumed that only 2 BlogEx exists in DbContext (1st BlogEx from test-blog-1)
+                entityChange.EntityId.ShouldBe(2.ToJsonString(false, false));
```

Comment:
Rather than a comment:

```diff
  // Therefore, BlogPromotion should follow Blog and have entity history.

+ const int blog2Id = 2;
+
  WithUnitOfWork(() =>
  {
      var blog2 = _blogRepository.Single(b => b.Name == "test-blog-2");
+     blog2.Id.ShouldBe(blog2Id);
      blog2.More.ShouldBeNull();
      blog2.More = new BlogEx() { BloggerName = "blogger-2" };
  });

  ...

  Predicate<EntityChangeSet> predicate = s =>
  {
      s.EntityChanges.Count.ShouldBe(1);

      var entityChange = s.EntityChanges.Single(ec => ec.EntityTypeFullName == typeof(BlogEx).FullName);
      entityChange.ChangeType.ShouldBe(EntityChangeType.Updated);
-     // The primary key of BlogEx is a shadow property,
-     // and we assumed that only 2 BlogEx exists in DbContext (2nd BlogEx from test-blog-2)
-     entityChange.EntityId.ShouldBe(2.ToJsonString(false, false));
+     entityChange.EntityId.ShouldBe(blog2Id.ToJsonString(false, false));
```

Reply:
This is incorrect.
`entityChange.EntityId.ShouldBe(2.ToJsonString(false, false));` is testing against the Id of BlogEx (not Blog)

Reply:
Are you sure?

Reply:
Yes, however, ef core maintains the PK of BlogEx to be the same as Blog. (I can change it to use BlogId still)

https://docs.microsoft.com/en-us/ef/core/modeling/owned-entities#implicit-keys

> In order to understand how EF Core tracks these objects, it is useful to know that a primary key is created as a shadow property for the owned type. The value of the key of an instance of the owned type will be the same as the value of the key of the owner instance.

Reply:
I see. The assumption is different for `OwnsMany`.
The comment "we assumed that only 2 BlogEx exists in DbContext" is also misleading.

Reply:
Done